Makefile.common

```diff
@@ -88,7 +88,13 @@ GOLANGCI_LINT_VERSION ?= v1.39.0
 # windows isn't included here because of the path separator being different.
 ifeq ($(GOHOSTOS),$(filter $(GOHOSTOS),linux darwin))
```

Comment:
probably we should align the version used. probably worth to upgrade to the v1.42.0 in the Makefile. 

I think we should specify the version also in the github action instead of letting it free. Without specifying, we won't control the change of the linter which we all know can change the rule and break a ci when upgrading the version :)

Reply:
Yup, that sounds like a good idea